compiler/qsc_rca/tests/qubits.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+pub mod common;
+
+use common::{check_last_statement_compute_properties, CompilationContext};
+use expect_test::expect;
+
+#[test]
+fn check_rca_for_static_single_qubit_allcation() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(r#"use q = Qubit();"#);
+    let package_store_compute_properties = compilation_context.get_compute_properties();
+    check_last_statement_compute_properties(
+        package_store_compute_properties,
+        &expect![
+            r#"
+            ApplicationsGeneratorSet:
+                inherent: Quantum: QuantumProperties:
+                    runtime_features: RuntimeFeatureFlags(0x0)
+                    value_kind: Element(Static)
+                dynamic_param_applications: <empty>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_static_multi_qubit_allcation() {
```

Comment:
Might be worth a test for dynamic qubit array, something like `use qs = Qubit[s];` where `s` is dynamic. Though given the transformations done by the passes, it might be hard to find the right statement to check the properties on, since it may not be the last one.

Reply:
Updated the tests to include a dynamic qubit array, as well as a dynamic single qubit.

Using the trick of checking an expression's properties through the variable it is binded too worked here as well.

For the dynamic single qubit allocation, I verified the behavior by encapsulating the dynamic qubit allocation in a callable.